pkg/istiod/istio_ca.go

```diff
@@ -0,0 +1,383 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istiod
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"google.golang.org/grpc/metadata"
+	"io/ioutil"
+	"istio.io/istio/security/pkg/server/ca/authenticate"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"istio.io/pkg/env"
+
+	"google.golang.org/grpc"
+	"k8s.io/client-go/kubernetes"
+
+	oidc "github.com/coreos/go-oidc"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	"istio.io/istio/pkg/spiffe"
+	"istio.io/istio/security/pkg/cmd"
+	"istio.io/istio/security/pkg/pki/ca"
+	caserver "istio.io/istio/security/pkg/server/ca"
+	"istio.io/pkg/log"
+)
+
+// Based on istio_ca main - removing creation of Secrets with private keys in all namespaces and install complexity.
```

Comment:
I'm not sure I understand the first sentence, maybe I'm misreading it. What does "install complexity" mean here?

Reply:
Setting up SDS and certs involved a lot of options and steps. In istiod we only support SDS (no Secrets), and most of the steps are no longer needed.